src/main/java/subway/line/service/LineService.java

```diff
@@ -76,4 +83,61 @@ private Line findLineById(final Long lindId) {
         return lineRepository.findById(lindId)
                 .orElseThrow(() -> new EntityNotFoundException("Line not found. Line Id: " + lindId));
     }
+
+    @Transactional
+    public LineResponse createLineSection(final Long lineId, final SectionRequest request) {
+        final Line line = this.findLineById(lineId);
+
+        if (this.isSectionRegistered(request, line)) {
+            throw new IllegalArgumentException();
+        }
+
+        this.validateNoDuplicateDownStation(request);
+
+        final Station requestUpStation = this.findStationById(request.getUpStationId());
+        final Station requestDownStation = this.findStationById(request.getDownStationId());
+
+        final Section section = new Section(line, requestUpStation, requestDownStation, request.getDistance());
+        sectionRepository.save(section);
+
+        line.addDownStation(requestDownStation, request.getDistance());
```

Comment:
유효성을 검증하는 로직과 노선에 구간을 추가하는 로직은 Line 내부로 이동하면 어떨까요?
모든 비즈니스 로직을 서비스가 아닌 도메인에서 수행하도록 개선해 보시면 좋을 것 같아요 
```suggestion
        final Station requestUpStation = this.findStationById(request.getUpStationId());
        final Station requestDownStation = this.findStationById(request.getDownStationId());

        final Section section = new Section(line, requestUpStation, requestDownStation, request.getDistance());
        line.addSection(section);
```

Reply:
유효성을 검증하는 로직은 비즈니스 로직에서 하는 것보단 도메인에서 수행하는 것이 일반적인가요?
기준을 제대로 세우지 못해서 매번 소스 코드를 작성할 때마다 어디에 위치하는 것이 옳은지 고민이었어요

Reply:
구간을 추가하는 책임은 도메인에 있는데 유효성 검증을 도메인에서 수행하지 않는다면 책임이 분리되었다고 볼 수 있을 것 같아요
즉 응집도가 낮은 구조라고 볼 수 있겠네요 
여러 관점이 있을 수 있겠지만 응집도 관점에서 유효성 검증을 어느 위치에 둘 지 고민해 보시면 좋을 것 같아요